### ISSUE TITLE
ethminer 0.18.0-rc.0 hangs at startup

### ISSUE THREAD

**Issue by astinus000** [ethereum-mining/ethminer#1879]
**Describe the bug**
When starting ethminer, the process freezes almost immediately.

**To Reproduce**
win 10x64
Latest AMD drivers 19.4.1
start process with ethminer.exe -v 5 --farm-recheck 200 -G --cl-local-work 256 --cl-global-work 16384 -P stratum://xxx.worker@server/pass

Reproduced on AMD RX470/580
here is the cut log
-----------
i 22:19:52 <unknown> Spinning up miners...
cl 22:19:52 cl-0     Using PciId : 01:00.0 Ellesmere OpenCL 2.0 AMD-APP (2766.5) Memory : 4.00 GB
cl 22:19:52 cl-1     Using PciId : 02:00.0 Ellesmere OpenCL 2.0 AMD-APP (2766.5) Memory : 4.00 GB
cl 22:19:52 cl-2     Using PciId : 03:00.0 Ellesmere OpenCL 2.0 AMD-APP (2766.5) Memory : 4.00 GB
cl 22:19:52 cl-3     Using PciId : 04:00.0 Ellesmere OpenCL 2.0 AMD-APP (2766.5) Memory : 4.00 GB
 i 22:19:52 cl-0     Adjusting CL work multiplier for 32 CUs. Adjusted work multiplier: 58
-----------
it hangs at this point.

devices (first 4 - RX470, 5th - RX580, 6th - RX470):
-----------
ethminer 0.18.0-rc.0
Build: windows/release/msvc

CUDA Error : No CUDA driver found
 Id Pci Id    Type Name                          CUDA SM  CL    Total Memory  Cl Max Alloc  Cl Max W.Grp
--- --------- ---- ----------------------------- ---- --- ----  ------------  ------------  ------------
  0 01:00.0   Gpu  Ellesmere                              Yes        4.00 GB       3.75 GB      256.00 B
  1 02:00.0   Gpu  Ellesmere                              Yes        4.00 GB       3.75 GB      256.00 B
  2 03:00.0   Gpu  Ellesmere                              Yes        4.00 GB       3.75 GB      256.00 B
  3 04:00.0   Gpu  Ellesmere                              Yes        4.00 GB       3.75 GB      256.00 B
  4 05:00.0   Gpu  Ellesmere                              Yes        4.00 GB       3.75 GB      256.00 B
  5 06:00.0   Gpu  Ellesmere                              Yes        4.00 GB       3.75 GB      256.00 B
-----------

rolling back to 0.17.1 resolves the problem.
using ''--cl-nobin'' option does not resolves problem on 0.18.0-rc.0

**Comment by lesjokolat** [comment]
hi re ordered things a bit make sure no extra spaces...

ethminer.exe -G -P stratum://xxx.worker@server/pass --cl-local-work 256 --cl-global-work 16384 -v 5 --farm-recheck 200 

https://github.com/ethereum-mining/ethminer/blob/master/docs/POOL_EXAMPLES_ETH.md see Special character section. See if pool you are trying to connect to is listed here.

**Comment by astinus000** [comment]
Changing params order makes no difference. Using Nanopool.

**Comment by lesjokolat** [comment]
then looking at your startup order you have wrong order in script.  match the example style and format

**Comment by astinus000** [comment]
I've tryed launch ethermine as you suggested. It also doesn't work with same result.
But any variant works with 0.17.1 version.

**Comment by lesjokolat** [comment]
do you have an inbuilt video card on motherboard?
 how much scratch disk space you have allocated?

**Comment by astinus000** [comment]
Yes, I have intel onboard video. It is a primary video adapter. Motherboard Biostar TB85 with 6 pci-e GPUs.
HDD is 1.5TB, page file 32GB.

**Comment by lesjokolat** [comment]
then you will have to exclude the in uilt video card by specifying just the other ones.

    --opencl-devices <0 1 ..n> Select which OpenCL devices to mine on.

**Comment by astinus000** [comment]
I provide list of video cards recognized by ethminer in first message (ethminer --list-devices).
There is no built-in video card and only AMD video cards. So, I couldn't exclude built-in video.

**Comment by BoriZka67** [comment]
https://github.com/ethereum-mining/ethminer/issues/1773#issue-390528541

I have the same problem on AMD RX588 and miner version 18.Х. Found the following solution - use ethminer 0.17.1 )))

**Comment by Marty8mf** [comment]
Hi, any news on this ? I've the same problem, miner stay stucked on "Adjusting CL work multiplier for 32 CUs. Adjusted work multiplier: 58" on all my RX 570 rigs.
Same problem on my PC with Vega 56.
On RX 580 no problem because there no need to adjust CL work multiplier.
W10 and drivers 18.6.1 on all my rigs.
Tried various driver versions on a test rig, always same problem.
Problem is other miners based on your code have same problem because they use current code version, like ZIL Miner for exemple.
Any help please ? Thanks
